sos/plugins/__init__.py

```diff
@@ -601,11 +723,16 @@ def get_option_as_list(self, optionname, delimiter=",", default=None):
     def _add_copy_paths(self, copy_paths):
         self.copy_paths.update(copy_paths)
 
-    def add_copy_spec(self, copyspecs, sizelimit=None, tailit=True):
+    def add_copy_spec(self, copyspecs, sizelimit=None, tailit=True, pred=None):
         """Add a file or glob but limit it to sizelimit megabytes. If fname is
         a single file the file will be tailed to meet sizelimit. If the first
         file in a glob is too large it will be tailed to meet the sizelimit.
         """
+        if not self.test_predicate(pred=pred):
+            self._log_info("skipped copy spec '%s' due to predicate" %
```

Comment:
Maybe for future debugging purposes, log also the "guilty" predicate? (i.e.:

    self._log_info("skipped copy spec '%s' due to predicate '%s'" %
                                (copyspecs, self.get_predicate(pred)))

)?